lib/3scale/backend/storage_async/client.rb

```diff
@@ -71,20 +72,34 @@ def initialize_client(opts)
         end
 
         def init_host_client(opts)
-          uri = URI(opts[:url] || '')
-          host = uri.host || DEFAULT_HOST
-          port = uri.port || DEFAULT_PORT
-
-          endpoint = Async::IO::Endpoint.tcp(host, port)
-          Async::Redis::Client.new(endpoint, limit: opts[:max_connections])
+          endpoint = make_redis_endpoint(opts)
+          protocol = make_redis_protocol(opts)
+          Async::Redis::Client.new(endpoint, protocol: protocol, limit: opts[:max_connections])
         end
 
         def init_sentinels_client(opts)
           uri = URI(opts[:url] || '')
           name = uri.host
           role = opts[:role] || :master
+          protocol = make_redis_protocol(opts)
+
+          Async::Redis::SentinelsClient.new(name, opts[:sentinels], role, protocol)
+        end
+
+        # RESP2 with support for logical DBs
+        def make_redis_protocol(opts)
+          uri = URI(opts[:url] || "")
+          db = uri.path[1..-1]
```

Comment:
I'm thinking whether we need to report an error if the `db` is not an integer.
Or just let the client respond to `select {invalid-value}` with whatever it likes?

Reply:
Yeah, I wouldn't bother reraising this.

Reply:
agree with @akostadinov